packages/compass-components/src/components/bson-value.tsx

```diff
@@ -0,0 +1,288 @@
+import React, { useMemo } from 'react';
+import { Binary } from 'bson';
+import type {
+  Code,
+  Double,
+  Int32,
+  Long,
+  BSONRegExp,
+  Timestamp,
+  DBRef,
+  MaxKey,
+  MinKey,
+  ObjectId,
+  BSONSymbol,
+  Decimal128,
+} from 'bson';
+import { css } from '..';
+
+type BSONValueProps =
+  | { type: 'Binary'; value: Binary }
```

Comment:
It might've make sense to just use `_bsontype` property of the value inside the component instead of requiring an explicit type manually passed, but I wasn't sure if this is a good idea (the property is part of the public interface but underscore kinda hints that maybe it should not be used?) or if it will cause any changes in behavior of the app, so decided to keep the interface as it was before for now

Reply:
That reassures me and makes me tempted to try this implementation out 😄